dbt/parser/schemas.py

```diff
@@ -61,186 +62,386 @@ def as_kwarg(key, value):
-                'kwargs': ", ".join(kwargs)
-            }
+        return test_namespace, test_type, package_name
+
+    @classmethod
+    def build_unparsed_node(cls, model_name, package_name, test_type,
+                            test_args, root_dir, original_file_path):
+        """Given a model name (for the model under test), a pacakge name,
+        a test type (identifying the test macro to use), arguments dictionary,
+        the root directory of the search, and the original file path to the
+        schema.yml file that specified the test, build an UnparsedNode
+        representing the test.
+        """
+        test_path = os.path.basename(original_file_path)
+        original_test_type = test_type
+        test_namespace, test_type, package_name = cls.calculate_namespace(
+            test_type, package_name
         )
 
-        base_path = test_base.get('path')
+        raw_sql = build_test_raw_sql(test_namespace, model_name, test_type,
+                                     test_args)
+
         hashed_name, full_name = get_nice_schema_test_name(test_type,
                                                            model_name,
                                                            test_args)
 
-        hashed_path = get_pseudo_test_path(hashed_name, base_path,
+        hashed_path = get_pseudo_test_path(hashed_name, test_path,
                                            'schema_test')
-        full_path = get_pseudo_test_path(full_name, base_path,
+        full_path = get_pseudo_test_path(full_name, test_path,
                                          'schema_test')
-
-        # supply our own fqn which overrides the hashed version from the path
-        fqn_override = cls.get_fqn(full_path, package_project_config)
-        package_name = test_base.get('package_name')
-        node_path = cls.get_path(NodeType.Test, package_name, full_name)
-
-        to_return = UnparsedNode(
+        return UnparsedNode(
             name=full_name,
-            resource_type=test_base.get('resource_type'),
+            resource_type=NodeType.Test,
             package_name=package_name,
-            root_path=test_base.get('root_path'),
+            root_path=root_dir,
             path=hashed_path,
-            original_file_path=test_base.get('original_file_path'),
+            original_file_path=original_file_path,
             raw_sql=raw_sql
         )
 
-        return cls.parse_node(to_return,
+    @classmethod
+    def build_parsed_node(cls, unparsed, model_name, root_project,
+                          all_projects, macros):
+        """Given an UnparsedNode with a node type of Test and some extra
+        information, build a ParsedNode representing the test.
+        """
+
+        test_path = os.path.basename(unparsed.original_file_path)
+
+        source_package = all_projects.get(unparsed.package_name)
+        if source_package is None:
+            desc = '"{}" test on model "{}"'.format(original_test_type,
```

Comment:
where are `original_test_type` and `test_namespace` coming from here?

Reply:
whoops. I'll fix this by rearranging `build_node` and friends.